packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -535,11 +537,15 @@ function getFormBodyAndHeaders(
 	return { body: postBody, headers: header };
 }
 
-export function evalBlobsAndTrees(snapshot: ISnapshotContents) {
+export function evalBlobsAndTrees(snapshot: ISnapshotContents | IPartialSnapshotWithContents) {
 	const trees = countTreesInSnapshotTree(snapshot.snapshotTree);
-	const numBlobs = snapshot.blobs.size;
+	const numBlobs = instanceOfISnapshotContents(snapshot)
```

Comment:
Same feedback as above - code like that should not need to change.
Can IPartialSnapshotWithContents inherit from ISnapshotContents? Maybe it's unfortunate that we called property "blobs" and it needs to be called "blobsContents", but the question on "should we rename it" should be orthogonal to how we do proper type dependencies here to avoid most of the "simple" code in repo to need to know nuances about new features.

Reply:
Refined this function and made IPartialSnapshotWithContents inherit from ISnapshotTree but lets keep IPartialSnapshotWithContents and ISnapshotContents separate as it would be complicated to mix these 3 types. Also, ISnapshotConents is just localized to odsp driver. So, it would have minimal impact.